File: server/Model/Style.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, ManyToMany, JoinTable } from "typeorm";
import { Edit } from "./Edit";
import { Editor } from "./Editor";
import { Audio } from "./Audio";

@Entity()
export class Style {
    @PrimaryGeneratedColumn()
    style_id: number;

    @OneToMany(() => Edit, edit => edit.edit_style)
    style_edits: Edit[];

    @ManyToMany(() => Editor, editor => editor.editor_styles)
    @JoinTable()
    style_editor: Editor[];

    @ManyToMany(() => Audio, audio => audio.audio_styles)
    @JoinTable()
    style_audios: Audio[];

    @Column()
    style_name: string;
}

File: server/src/main/java/laurriana/time2day/model/Style.java
package laurriana.time2day.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Style {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int styleId;

    private String styleName;

    @OneToMany
    private ArrayList<Edit> styleEdits;

    @ManyToMany
    private ArrayList<Editor> styleEditors;

    @ManyToMany
    private ArrayList<Audio> styleAudios;
}
